Code/DataStructs/BitOps.cpp

```diff
@@ -137,12 +146,16 @@ bool AllProbeBitsMatch(const T1& probe, const std::string& pkl) {
   text += sizeof(size);
   if (size < 0) {
     version = -1 * size;
-    if (version == 16) {
-      format = 1;
-    } else if (version == 32) {
-      format = 2;
-    } else {
-      throw("Unknown version type for the encode bit vect");
+    switch (version) {
+      case 16:
+        format = 1;
+        break;
+      case 32:
+        format = 2;
+        break;
+      default:
+        throw("Unknown version type for the encode bit vect");
```

Comment:
It would be good to switch this to a ValueErrorException like what you have below in BitVect.cpp